impeller/renderer/backend/vulkan/command_buffer_vk.cc

```diff
@@ -47,6 +48,48 @@ const std::shared_ptr<CommandEncoderVK>& CommandBufferVK::GetEncoder() const {
   return encoder_;
 }
 
+bool CommandBufferVK::SubmitCommandsAsync(std::shared_ptr<BlitPass> blit_pass) {
+  TRACE_EVENT0("impeller", "CommandBufferVK::SubmitCommandsAsync");
+  if (!blit_pass->IsValid() || !IsValid()) {
+    return false;
+  }
+  auto context = context_.lock();
+  if (!context) {
+    return false;
+  }
+  if (!blit_pass->EncodeCommands(context->GetResourceAllocator())) {
+    return false;
+  }
+
+  const auto& context_vk = ContextVK::Cast(*context);
+  if (!encoder_->Finish()) {
+    return false;
+  }
+  context_vk.GetCommandBufferQueue()->Enqueue(std::move(encoder_));
+  return true;
+}
+
+bool CommandBufferVK::SubmitCommandsAsync(
+    std::shared_ptr<RenderPass> render_pass) {
+  TRACE_EVENT0("impeller", "CommandBufferVK::SubmitCommandsAsync");
+  if (!render_pass->IsValid() || !IsValid()) {
```

Comment:
nit: these can be the same function instead of duplicating the logic

```c++
namespace {
  template<typename Pass, typename Context, typename Encoder, typename EncodeFunc>
  {
    bool SubmitCommandsAsync(Pass pass, Context, Encoder encoder, Encoder encode_func) {
    // ...
   }
}
```